File: c言語/基礎知識用の勉強/2配列勉強/配列.c
#include <stdio.h>
#include <string.h>

int main(void)
{
	int asakuno[5];
	int asa[5][3] = {0};
	int kuno[] = {0, 1, 2, 3, 4};
	int aaa[][2] = {0, 1, 2, 3, 4};
	char tomohiro[16] = {0};
	char tomo[16] = "朝来野智博";

	// 以下、ポインタ配列(アルゴリズムの勉強では使わない？)。
	char *pasa[3] = {"朝来野智博", "朝来野", "智博"};
	char asaarray[][16] = {
		"朝来野智博",
		"朝来野",
		"智博",
	};

	asakuno[0] = 20210903;
	asakuno[1] = asakuno[0] + 1;
	asakuno[2] = asakuno[0] + 2;
	asakuno[3] = asakuno[0] + 3;
	asakuno[4] = asakuno[0] + 4;

	// 漢字1文字3バイト？
	strcpy(tomohiro, "朝来野智博");

	for ( int ii = 0; ii < 5; ii++ ) {
		for ( int jj = 0; jj < 3; jj++ ) {
			printf("asa = %d\n", asa[ii][jj]);	// 全て0で初期化されている。
		}
	}

	printf("asakuno = %d\n", asakuno[0]);	// asakuno = 20210903
	printf("asakuno = %d\n", asakuno[1]);	// asakuno = 20210904
	printf("asakuno = %d\n", asakuno[2]);	// asakuno = 20210905
	printf("asakuno = %d\n", asakuno[3]);	// asakuno = 20210906
	printf("asakuno = %d\n", asakuno[4]);	// asakuno = 20210907

	printf("asakuno配列の大きさ%ld\n", sizeof(asakuno)/sizeof(asakuno[0]));	// asakuno配列の大きさ5

	// 以下、2次元配列の大きさ確認
	printf("intの大きさ%ld\n", sizeof(int));	// intの大きさ4
	printf("asa配列の大きさ%ld\n", sizeof(asa));	// asa配列の大きさ60	←☆4*5*3=60
	printf("asa配列の大きさ%ld\n", sizeof(asa)/sizeof(asa[0]));	// asa配列の大きさ5	←☆asa[*]の大きさ。
	printf("asa配列の大きさ%ld\n", sizeof(asa[0])/sizeof(asa[0][0]));	// asa配列の大きさ3	←☆asa[][*]の大きさ。

	printf("tomohiro = %s\n", tomohiro);	// tomohiro = 朝来野智博
	printf("tomo = %s\n", tomo);	// tomo = 朝来野智博


	// 以下、ポインタ配列。
	printf("pasa = %s\n", pasa[0]);	// pasa = 朝来野智博
	printf("pasa = %s\n", pasa[1]);	// pasa = 朝来野
	printf("pasa = %s\n", pasa[2]);	// pasa = 智博

	printf("asaarray = %s\n", asaarray[0]);	// asaarray = 朝来野智博
}

// vim: set ts=4 sts=4 sw=4 tw=0 ff=unix fenc=utf-8 ft=c noexpandtab:

File: Perl言語/基礎知識用の勉強/2配列勉強/配列.pl
use v5.24;

my @asakuno;

print @asakuno;	# 出力なし。
@asakuno = (20210831, 20210901, );
print @asakuno;	# 2021083120210901

# 要素指定の取り出し。
say;
say "-" x 30;
say $asakuno[0];	# 20210831
say $asakuno[1];	# 20210901
say "-" x 30;

# 最後の要素取り出し。
say $asakuno[-1];	# 20210901
say $asakuno[-2];	# 20210831	←☆最後から2番目の要素。

my $count_asakuno = @asakuno;
say $count_asakuno;	# 2

say "-" x 30;
# 計算結果の代入。
say "以下、計算結果の代入。";
$asakuno[9] = $asakuno[1] + 9;
say $asakuno[8];	# 空
say $asakuno[9];	# 20210910

say "-" x 30;

# 最後の要素取り出し。
say $asakuno[-1];	# 20210910
say $#asakuno;	# 9
say $asakuno[$#asakuno];	# 20210910
say $asakuno[10];	# 空undef

say "-" x 30;


# 1つづつ取り出し。
foreach my $tomohiro (@asakuno) {
	say $tomohiro;	# 20210831
					# 20210901
					#
					#
					#
					#
					#
					#
					#
					# 20210910
}
say @asakuno;	# 202108312021090120210910

# 存在しない配列番号の要素出力。
say "-";
say $asakuno[99];	# 空行が出力されるようだ。
say "-";

@asakuno[999] = (20210901 + 9, );
say $asakuno[999];	# 20210910
@asakuno[999] = 20210901 + 9;
say $asakuno[999];	# 20210910
$asakuno[8] = 'asakuno';
$asakuno[7] = "tomohiro";
say $asakuno[8];	# asakuno
say $asakuno[7];	# tomohiro

# qw関数による代入。
my @asakuno = qw(a s a k u n o , );
say @asakuno;	# asakuno

foreach (@asakuno) {
	say $_;	# a
			# s
			# a
			# k
			# u
			# n
			# o
}

# ダブル(シングル)クォーテーションで括る。
say "@asakuno";	# a s a k u n o
say '@asakuno';	# @asakuno

# 文字列の@記号出力
#say "asakuno@tomohiro";	# Execution of 配列.pl aborted due to compilation errors.
say "asakuno\@tomohiro";	# asakuno@tomohiro
say 'asakuno\@tomohiro';	# asakuno\@tomohiro

# 要素数
my $len = @asakuno;
say $len;	# 8

say scalar(@asakuno); # 8

# 配列要素の空確認。
if (@asakuno > 0) {
	say "空でない。"
}
# 別の方法。
if (@asakuno) {
	say "空でない。"
}

my @asakuno;
if (@asakuno > 0) {
	say "空でない。";
}
else {
	say "空。";
	say scalar(@asakuno);	# 0		要素数がないため、0になる。
	say $#asakuno;	# -1		要素数がないため、最後の添え字は-1になる。
}

# 小数点以下のリストからの添え字利用。
@asakuno = (20210831, 20210901, );
my $tomohiro = 1.1234;
say $asakuno[$tomohiro];	# 20210901
#	これは、小数点数が切り捨てられて添え字として利用されているため、2つ目の要素(添え字1)が取り出された。
say $asakuno[$#asakuno];	# 20210901
say $#asakuno;	# 1

say "以上。"
# vim: set ts=4 sts=4 sw=4 tw=0 ff=unix fenc=utf-8 ft=perl noexpandtab:
